gdk/build_system/Zip.py

```diff
@@ -38,10 +39,26 @@ def build(self, **kwargs):
             artifacts_zip_build = Path(zip_build).joinpath(utils.get_current_directory().name).resolve()
             utils.clean_dir(zip_build)
             logging.debug("Copying over component files to the '{}' folder.".format(artifacts_zip_build.name))
+
+            def ignore_patterns_in_root(rootdir, *patterns):
```

Comment:
What is the purpose of the nested method definition here? Focus on readability. It might work, but is there a simpler way that you can write it?

Reply:
Was to return the callable while keeping the same overall syntax/implementation as the shutils base implementation, but I've refactored it to be a lot more readable in the latest commit.